homeUi/src/main/kotlin/com/gravatar/app/homeUi/presentation/home/share/components/ShareEditableField.kt

```diff
@@ -0,0 +1,91 @@
+package com.gravatar.app.homeUi.presentation.home.share.components
+
+import androidx.annotation.StringRes
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.BasicTextField
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Switch
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.SolidColor
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.gravatar.app.homeUi.R
+
+@Composable
+internal fun ShareEditableField(
+    @StringRes placeholder: Int,
+    value: String,
+    onValueChange: (String) -> Unit,
+    switchChecked: Boolean,
+    onSwitchCheckedChange: (Boolean) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Row(
+        modifier = modifier.fillMaxWidth(),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        BasicTextField(
+            value = value,
+            onValueChange = onValueChange,
+            textStyle = MaterialTheme.typography.bodyLarge.copy(color = MaterialTheme.colorScheme.onSurface),
+            singleLine = true,
+            cursorBrush = SolidColor(MaterialTheme.colorScheme.primary),
+            modifier = Modifier.weight(1f),
+            decorationBox = { innerTextField ->
+                Box(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .background(MaterialTheme.colorScheme.surfaceContainerHigh, RoundedCornerShape(8.dp))
+                        .padding(horizontal = 12.dp, vertical = 12.dp)
+                ) {
+                    if (value.isEmpty()) {
+                        Text(
+                            text = stringResource(placeholder),
+                            style = MaterialTheme.typography.bodyLarge,
+                            color = MaterialTheme.colorScheme.onSurfaceVariant
+                        )
+                    }
+                    innerTextField()
+                }
+            }
+        )
+        Switch(
+            checked = switchChecked,
+            onCheckedChange = onSwitchCheckedChange,
+            modifier = Modifier.padding(start = 8.dp)
```

Comment:
```suggestion
            modifier = Modifier.padding(start = 16.dp)
```

It should be 16 based on Figma.

Reply:
💡 Also, you can achieve this by having `horizontalArrangement = Arrangement.spacedBy(16.dp)` in the Row component.